### PR TITLE
Move DOMParser to HTML

### PR DESCRIPTION
**Pull request by domenic**
Follows https://github.com/whatwg/html/pull/5190.

Should only be merged after https://github.com/whatwg/html/pull/5190.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -205,88 +205,16 @@ <h2>Extensibility</h2>
...
-      <li>Return the value of <var>document</var>.
-    </ol>
-  </dl>
-
-  <p>In any case, the returned <a>Document</a>'s <a>content type</a> must be the <var>type</var>
-  argument. Additionally, the <a>Document</a> must have a <a>URL</a> value equal to the URL of the
-  <a>active document</a>, and a <a>location</a> value of <code>null</code>.
-
-  <p class=note>The returned <a>Document</a>'s <a>encoding</a> is the default, UTF-8.
+  <p>The definition of <code>DOMParser</code> has moved to <a href="https://multipage.html.spec.whatwg.org/dynamic-markup-insertion.html#domparser">the HTML Standard</a>.</p>
```
Comment (sideshowbarker) [inline index.html]:
```suggestion
  <p>The definition of <code>DOMParser</code> has moved to <a href="https://html.spec.whatwg.org/multipage/dynamic-markup-insertion.html#domparser">the HTML Standard</a>.</p>
```

File: index.html [hunk 2]
```diff
@@ -205,88 +205,16 @@ <h2>Extensibility</h2>
...
-    exception is thrown [[WEBIDL]].
-  </dl>
+  <span id="dom-domparser"></span>
+  <span id="dom-domparser-parsefromstring"></span>
+  <span id="idl-def-supportedtype"></span>
+  <span id="dom-supportedtype-text/html"></span>
+  <span id="dom-supportedtype-text/xml"></span>
+  <span id="dom-supportedtype-application/xml"></span>
+  <span id="dom-supportedtype-application/xhtml+xml"></span>
+  <span id="dom-supportedtype-image/svg+xml"></span>
```
Comment (sideshowbarker) [inline index.html]: This mechanism for preserving the IDs is clever. We should do the same thing for any other cases where we’re moving something out of a spec, and maybe consider documenting it somewhere as a best practice (though I don’t know often this case comes up). Among other reasons, it ensures any existing links in the Specifications tables in MDN articles still bring readers to the point in the document where they need to be (in this, to the statement that the content’s been moved to the HTML spec).

File: index.html (hunk 1 above)
Comment (domenic) [reply index.html]: Thank you for catching this!!

File: index.html (hunk 2 above)
Comment (domenic) [reply index.html]: We've been doing it in HTML when moving things around, and Bikeshed has `oldids=""` (https://tabatkins.github.io/bikeshed/#id-gen). There might even be something similar in ReSpec, but I'm not as familiar...

**Pull request merged**